src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -1082,6 +1086,103 @@ public unsafe void CopyFromAscii(string data)
             _index = blockIndex;
         }
 
+        private static byte[] NumericBytesScratch => _numericBytesScratch ?? CreateNumericBytesScratch();
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static byte[] CreateNumericBytesScratch()
+        {
+            var bytes = new byte[_maxULongByteLength];
+            _numericBytesScratch = bytes;
+            return bytes;
+        }
+
+        public unsafe void CopyFromNumeric(ulong value)
+        {
+            const byte AsciiDigitStart = (byte)'0';
+
+            var block = _block;
+            if (block == null)
+            {
+                return;
+            }
+
+            var blockIndex = _index;
+            var bytesLeftInBlock = block.Data.Offset + block.Data.Count - blockIndex;
+            var start = block.DataFixedPtr + blockIndex;
+
+            if (value < 10)
+            {
+                if (bytesLeftInBlock < 1)
+                {
+                    goto overflow;
```

Comment:
Is this that much better than just calling `CopyFromNumericOverflow(value);` and returning? I prefer avoiding goto if we can.